keps/sig-node/4009-add-cdi-devices-to-device-plugin-api/kep.yaml

```diff
@@ -0,0 +1,44 @@
+title: Add CDI devices to device plugin API
+kep-number: 4009
+authors:
+  - "@elezar"
+owning-sig: sig-node
+participating-sigs: []
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+creation-date: 2023-05-16
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+see-also:
+  - "/keps/sig-node/3063-dynamic-resource-allocation"
+  - "/keps/sig-node/3573-device-plugin"
+replaces: []
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha|beta|stable
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: "v1.28"
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: "v1.28"
+  beta: "v1.29"
+  stable: "v1.30"
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: MyFeature
+    components:
+      - kube-apiserver
+      - kube-controller-manager
```

Comment:
Component should be `kubelet` here.

Reply:
Will update. Is the devicemanager a component in this context, or are these only higher-level k8s components?

Reply:
Also not, in the current implemention it is not possible to disable this -- except in the sense that the device plugn API may not be used.

Reply:
> Is the devicemanager a component in this context, or are these only higher-level k8s components?
We have to mention only higher-level k8s components here.

Reply:
Found it: https://github.com/kubernetes/enhancements/pull/1884 !!
Here we had added `cpu_ids` field in `ContainerResources` and `TopologyInfo` in `ContainerDevices` and never had a separate feature gate because the API was stable. Perhaps we can take the same approach. This is worth explicitly clarifying with the SIG and the production readiness reviewers.

Reply:
@swatisehgal that should have been feature gated!